### DISCUSSION TITLE
What api is needed to implement in the red range

### DISCUSSION THREAD

**Discussion by dddssw** [microsoft/vscode-discussions#1529] [Extension Development QnA] [answered]
![image](https://github.com/user-attachments/assets/81119844-352c-4c60-8a08-6b224ae5fbdf)

**Comment by gjsjohnmurray** [comment, marked as answer]
These are [CodeLenses](https://code.visualstudio.com/api/references/vscode-api#CodeLens)

**Comment by dddssw** [comment]
Thank very much
